components/atom/tooltip/src/index.js

```diff
@@ -0,0 +1,234 @@
+      })
+    })
+  }
+
+  componentDidMount() {
+    const target = this.refTarget.current
+    this.props.innerRef(target)
+    document.addEventListener('click', this.handleClickOutsideElement)
+    target.oncontextmenu = this.handleContextMenu
+    target.addEventListener('mouseover', this.disableTitle)
+    target.addEventListener('mouseout', this.restoreTitle)
+  }
+
+  componentWillUnmount() {
+    const target = this.refTarget.current
+    clearTimeout(this.touchTimer)
+    document.removeEventListener('click', this.handleClickOutsideElement)
+    this.refTarget.removeEventListener('touchend', this.toggle)
+    target.removeEventListener('mouseover', this.disableTitle)
+    target.removeEventListener('mouseout', this.restoreTitle)
+  }
+
+  disableTitle(e) {
+    this.dataset.title = this.title
+    this.title = ''
+  }
+
+  restoreTitle(e) {
+    this.title = this.dataset.title
+  }
+
+  handleContextMenu = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    return false
+  }
+
+  handleClickOutsideElement = e => {
+    const {isOpen} = this.state
+    if (isOpen) {
+      const tooltipDom = this.refTooltip.current
+      const isOutside = tooltipDom && !tooltipDom.contains(e.target)
+      if (isOutside) this.toggle(e)
+    }
+  }
+
+  handleTouchStart = e => {
+    this.preventNonTouchEvents = true
+    this.hasTouchEnded = false
+    clearTimeout(this.touchTimer)
+    this.touchTimer = setTimeout(() => {
+      if (!this.hasTouchEnded) {
+        this.setState({
+          isOpen: !this.state.isOpen
+        })
+      }
+      this.preventNonTouchEvents = false
+      this.hasTouchEnded = false
+    }, 1000)
+    return false
+  }
+
+  handleTouchEnd = e => {
+    if (!this.preventNonTouchEvents) this.handleStopPropagation(e)
+    this.hasTouchEnded = true
+    clearTimeout(this.touchTimer)
+  }
+
+  handleStopPropagation = e => {
+    e.preventDefault()
+    e.stopPropagation()
+    return false
+  }
+
+  toggle = e => {
+    const {type} = e
+    if (this.onClickTarget) {
+      if (type === 'touchstart') this.handleTouchStart(e)
+      if (type === 'touchend') this.handleTouchEnd(e)
+      if (type === 'click') this.onClickTarget(e)
+      if (!this.preventNonTouchEvents) {
+        this.setState({
+          isOpen: !this.state.isOpen
+        })
+      }
+    } else {
+      if (type === 'touchstart') this.hasTouchEnded = false
+      if (type === 'touchend') this.hasTouchEnded = true
+      if (this.hasTouchEnded && ['focusin', 'mouseover'].includes(type)) {
+        this.handleStopPropagation(e)
+      } else {
+        if (['touchstart', 'touchend'].includes(type)) return
+        if (['click', 'focusin'].includes(type)) this.handleStopPropagation(e)
+        this.setState({
+          isOpen: !this.state.isOpen
+        })
+      }
+    }
+  }
+
+  render() {
+    const {hideArrow, delay, autohide, placement} = this.props // eslint-disable-line react/prop-types
+    const target = this.refTarget.current
```

Comment:
If I don't remember wrong, you aren't able to cache this as it changes? Maybe you could add a comment explaining why